bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergenieSocket.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.energenie.internal.handler;
+
+import static org.openhab.binding.energenie.internal.EnergenieBindingConstants.*;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class handling the Socket connections.
+ *
+ * @author Hans-Jörg Merk - Initial contribution
+ * @author Hilbrand Bouwkamp - Moved Socket to it's own class
+ */
+@NonNullByDefault
+class EnergenieSocket {
+
+    private static final int SOCKET_TIMEOUT_MILLISECONDS = 1500;
+    private static final byte[] MESSAGE = { 0x11 };
+
+    private final Logger logger = LoggerFactory.getLogger(EnergenieSocket.class);
+    private final String host;
+    private final byte[] key;
+
+    public EnergenieSocket(final String host, final String password) {
+        this.host = host;
+        key = getKey(password);
```

Comment:
Just for consistency.
```suggestion
        this.key = getKey(password);
```

Reply:
Done